### ISSUE TITLE
Newer version of glibc to support Recognize WASM mode

### ISSUE THREAD

**Issue by mr-bryn** [nextcloud-snap/nextcloud-snap#2808]
I have been running the Recognize app successfully within NC snap, but recently it stopped working.

Based on [the discussion over there](https://github.com/nextcloud/recognize/issues/1130) it would seem the version of tensorflow used requires a specific version of node.js, which itself requires at least GLIBC 2.28.

From what I can determine my system (Debian 12) has GLIBC 2.36, but within the snap environment it reports GLIBC 2.27

I really don't know enough about snaps. Is there a way to update glibc to a newer version or use the version on my system?

**Comment by pachulo** [comment]
> I really don't know enough about snaps. Is there a way to update glibc to a newer version or use the version on my system?

No. The proper way is [to upgrade the snap so it uses uses a newer core](https://github.com/nextcloud-snap/nextcloud-snap/issues/2011), but we still don't have any ETA or started working on that, sorry.

**Comment by mr-bryn** [comment]
> The proper way

Does this mean there is an 'improper' way until the new core is used?

**Comment by pachulo** [comment]
>> The proper way

> Does this mean there is an 'improper' way until work to use the new core is done?


Not that I know of

**Comment by scubamuc** [comment]
@mr-bryn, you may find some useful information referring to upgrade issues with "recognize" in our [FAQ's](https://github.com/nextcloud-snap/nextcloud-snap/wiki/FAQ's#q-nextcloud-snap-is-not-working-because-of-third-party-app-incompatibility-what-can-i-do)

especially:

   > **INFO**: Recognize app path to `node.js` is reset by the app after an Nextcloud snap update. The admin would need to configure the path to be `/var/snap/nextcloud/current/nextcloud/extra-apps/recognize/bin/node`.
   
   > **Workaroud**: uninstall Recognize app before an update and reinstall after update -- if the Snap will not start after update, [revert](https://github.com/nextcloud-snap/nextcloud-snap/wiki/Managing-Nextcloud-snap-with-Snap#revert-nextcloud-snap) and remove Recognize, then [update manually](https://github.com/nextcloud-snap/nextcloud-snap/wiki/Managing-Nextcloud-snap-with-Snap#update-nextcloud-snap) and reinstall Recognize

**Comment by carlosgalveias** [comment]
Any news? I love the Recignize and snap

**Comment by pachulo** [comment]
@carlosgalveias this is still work-in-progress https://github.com/nextcloud-snap/nextcloud-snap/pull/3073

**Comment by pachulo** [comment]
Hi! We are now testing [the upgrade to core20](https://github.com/nextcloud-snap/nextcloud-snap/issues/2011), which means that [glib version 2.31 should be now available ](https://gist.github.com/richardlau/6a01d7829cc33ddab35269dacc127680) inside the snap.
You can help us testing if this works by upgrading to `latest/beta` with:
```
sudo snap refresh --beta nextcloud
```

**Comment by pachulo** [comment]
I tried installing recognize and the Node.js version that it downloaded seems to work fine:
```
$ sudo snap run --shell nextcloud.occ
root@nextcloud-container:/home/ubuntu# /var/snap/nextcloud/current/nextcloud/extra-apps/recognize/bin/node -v
v20.9.0
```
(I'm on a x64 system, so no WASM needed)
